src/main/java/com/minecolonies/core/colony/buildings/modules/BuildingResourcesModule.java

```diff
@@ -336,7 +336,13 @@ public void checkOrRequestBucket(@Nullable final BuilderBucket requiredResources
             return;
         }
 
-        resourceloop:
+        final ImmutableList<IRequest<? extends Stack>> list = building.getOpenRequestsOfType(-1, TypeToken.of(Stack.class));
+        final Set<ItemStorage> requestedItems = new HashSet<>();
+        for (final IRequest<? extends Stack> request : list)
+        {
+            requestedItems.add(new ItemStorage(request.getRequest().getStack()));
```

Comment:
why create a new itemstorage when it is only used for checking contains? that should be only needed if you need to modify it, should be able to just add to the existing ones from the list